internal/messenger/email/email.go

```diff
@@ -99,10 +113,17 @@ func (e *Emailer) Push(m messenger.Message) error {
 		ln  = len(e.servers)
 		srv *Server
 	)
-	if ln > 1 {
-		srv = e.servers[rand.Intn(ln)]
-	} else {
-		srv = e.servers[0]
+
+	for {
+		if ln > 1 {
+			srv = e.servers[rand.Intn(ln)]
+			if (srv.limiter.Allow()) {
+				break;
+			}
+		} else {
+			srv = e.servers[0]
```

Comment:
A limiter token needs to be consumed here when only 1 server is configured with a rate limit.